frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_projects.py

```diff
@@ -78,201 +115,384 @@ def owner2tuple(ownername):
+class ProjectEdit(Resource):
+    @staticmethod
+    def _common(ownername, projectname):
+        copr = get_copr(ownername, projectname)
+        data = rename_fields(get_form_compatible_data(preserve=["chroots"]))
+        form = forms.CoprForm(data, meta={"csrf": False})
 
-@apiv3_ns.route("/project/delete/<ownername>/<projectname>", methods=DELETE)
-@api_login_required
-def delete_project(ownername, projectname):
-    copr = get_copr(ownername, projectname)
-    copr_dict = to_dict(copr)
-    form = forms.APICoprDeleteForm(meta={'csrf': False})
+        if not form.validate_on_submit():
+            raise InvalidForm(form)
+        validate_chroots(get_input_dict(), MockChrootsLogic.get_multiple())
+
+        for field in form:
+            if field.data is None or field.name in ["csrf_token", "chroots"]:
+                continue
+            if field.name not in data.keys():
+                continue
+            setattr(copr, field.name, field.data)
+
+        if form.chroots.data:
+            CoprChrootsLogic.update_from_names(flask.g.user, copr, form.chroots.data)
 
-    if form.validate_on_submit() and copr:
         try:
-            ComplexLogic.delete_copr(copr)
-        except (ActionInProgressException,
-                InsufficientRightsException) as err:
+            CoprsLogic.update(flask.g.user, copr)
+            if copr.group:  # load group.id
+                _ = copr.group.id
+            db.session.commit()
+        except (
+            ActionInProgressException,
+            InsufficientRightsException,
+            NonAdminCannotDisableAutoPrunning,
+        ) as ex:
             db.session.rollback()
-            raise err
+            raise ex
+
+        return to_dict(copr)
+
+    @restx_api_login_required
+    @apiv3_projects_ns.doc(fullname_docs)
+    @apiv3_projects_ns.marshal_with(project_model)
+    @apiv3_projects_ns.expect(project_edit_input_model)
+    @apiv3_projects_ns.response(HTTPStatus.OK.value, "Copr project successfully edited")
+    @apiv3_projects_ns.response(
+        HTTPStatus.BAD_REQUEST.value, HTTPStatus.BAD_REQUEST.description
+    )
+    def put(self, ownername, projectname):
+        """
+        Edit Copr project
+        Edit existing Copr project for ownername/projectname in form.
+        """
+        return self._common(ownername, projectname)
+
+    @restx_api_login_required
+    @apiv3_projects_ns.doc(fullname_docs)
+    @apiv3_projects_ns.marshal_with(project_model)
+    @apiv3_projects_ns.expect(project_edit_input_model)
+    @apiv3_projects_ns.response(HTTPStatus.OK.value, "Copr project successfully edited")
+    @apiv3_projects_ns.response(
+        HTTPStatus.BAD_REQUEST.value, HTTPStatus.BAD_REQUEST.description
+    )
+    @deprecated_route_method_type(apiv3_projects_ns, "POST", "PUT")
+    def post(self, ownername, projectname):
+        """
+        Edit Copr project
+        Edit existing Copr project for ownername/projectname in form.
+        """
```

Comment:
It would be nice to warn users to use the proper HTTP method - e.g. PUT and fix it in python-copr.

I think mentioning it somewhere (e.g. release notes) is a good idea.

> we are going to deprecate this in the future and keep only PUT

I think a lot of users use our API in this "bad" way and I don't want to force them to fix their already working scripts. This idea is mainly about documentation saying it is deprecated and a warning header saying it is deprecated but no plan to drop the functionality whatsoever